src/JuliennedArrays.jl

```diff
@@ -142,26 +146,44 @@ julia> input = reshape(1:8, 2, 2, 2)
  6  8
 
 julia> s = Slices(input, 1, 3)
-2-element Slices{SubArray{Int64,2,Base.ReshapedArray{Int64,3,UnitRange{Int64},Tuple{}},Tuple{Base.OneTo{Int64},Int64,Base.OneTo{Int64}},false},1,Base.ReshapedArray{Int64,3,UnitRange{Int64},Tuple{}},Tuple{True,False,True}}:
+2-element Slices{SubArray{$Int, 2}, 1}:
  [1 5; 2 6]
  [3 7; 4 8]
 
 julia> map(sum, s)
-2-element Array{Int64,1}:
+2-element Vector{$Int}:
  14
  22
 ```
 """
-Slices(
-    whole::AbstractArray{Item,NumberOfDimensions},
-    alongs::Int...,
-) where {Item,NumberOfDimensions} =
-    Slices(whole, in_unrolled(as_vals(alongs...), ntuple(Val, NumberOfDimensions)...)...)
-export Slices
+function Slices(whole::AbstractArray{T,N}, alongs::Int...) where {T,N}
+    any(x->x>N, alongs) && throw(ArgumentError("All alongs values $(alongs) should be less than $(N)"))
+    Slices(whole, in_unrolled(as_vals(alongs...), ntuple(Val, N)...)...)
+end
+
+function Base.showarg(io::IO, ::Slices{T,N}, toplevel) where {T,N}
+    print(io, "Slices{", basetype(T), "{", eltype(T), ", ", ndims(T), "}, ", N, "}")
+end
 
+# This is expected to be added to Julia (maybe under a different name)
+# Follow https://github.com/JuliaLang/julia/issues/35543 for progress
+basetype(T::Type) = Base.typename(T).wrapper
+basetype(T) = basetype(typeof(T))
+
+###
+# Align
+###
 struct Align{Item,Dimensions,Sliced,Alongs} <: AbstractArray{Item,Dimensions}
     slices::Sliced
     alongs::Alongs
+    function Align{T,N,S,A}(slices::S, alongs::A) where {T,N,S,A}
+        sz = size(first(slices))
+        all(x->sz==size(x), slices) || throw(ArgumentError("All sizes of slices should be the same."))
+        length(alongs) == N || throw(DimensionMismatch("The total dimension $(N) is expected to be the sum of inner dimension $(length(sz)) and outer dimension $(length(alongs))"))
+        inner_dimensions = mapreduce(isequal(True()), +, alongs)
```

Comment:
It would be good to verify that these checks can be done by constant-propagation at compile-time

Reply:
Mostly, check the benchmark for `Align` https://github.com/bramtayl/JuliennedArrays.jl/pull/28#issuecomment-905222481